### PR TITLE
E2ETests: allow overriding the version of FRR used

### PR DESCRIPTION
**Pull request by fedepaol**
This to allow using the test suite against different deployments (i.e. on a real cluster) where there might be incompatibility across versions.

### CODE REVIEW

File: e2etest/e2etest_suite_test.go [hunk 1]
```diff
@@ -82,6 +83,7 @@ func handleFlags() {
 	flag.StringVar(&prometheusNamespace, "prometheus-namespace", "monitoring", "the namespace prometheus is running in (if running)")
 	flag.StringVar(&externalContainers, "external-containers", "", "a comma separated list of external containers names to use for the test. (valid parameters are: ibgp-single-hop / ibgp-multi-hop / ebgp-single-hop / ebgp-multi-hop)")
 	flag.BoolVar(&bgpNativeMode, "bgp-native-mode", false, "says if we are testing against a deployment using bgp native mode")
+	flag.StringVar(&frrImage, "frrimage", "quay.io/frrouting/frr:8.4.2", "says if we are testing against a deployment using bgp native mode")
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply e2etest/e2etest_suite_test.go:86]: done

[the remark this answers is not in the document]
Comment (liornoy) [reply e2etest/e2etest_suite_test.go:86]: should we keep the naming convention consistent with: "frr-image" as the flag?

[the remark this answers is not in the document]
Comment (danielmellado) [reply e2etest/e2etest_suite_test.go:86]: Shouldn't this just be along the lines of 'frr image version to be used'? Also worth mentioning in some docs which frr image goes along with metallb image.

File: tasks.py [hunk 2]
```diff
@@ -742,7 +742,7 @@ def helmdocs(ctx, env="container"):
     "external_containers": "a comma separated list of external containers names to use for the test. (valid parameters are: ibgp-single-hop / ibgp-multi-hop / ebgp-single-hop / ebgp-multi-hop)",
     "native_bgp": "tells if the given cluster is deployed using native bgp mode ",
```
Comment (zeeke) [inline tasks.py:743]: Please, add some documentation for this new parameter:

```suggestion
    "native_bgp": "tells if the given cluster is deployed using native bgp mode ",
    "frr_image": "specifies which FRR container should be used in e2e tests when creating external router endpoints. Useful to test compatibility with multiple FRR versions.",
```

I'd also change it to `external_containers_frr_image_for`, as it doesn't change in any way the FRR containers MetalLB uses for its work. Right?

Comment (fedepaol) [reply tasks.py:743]: right

Comment (fedepaol) [reply tasks.py:743]: done

File: e2etest/e2etest_suite_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fedepaol) [reply e2etest/e2etest_suite_test.go]: > should we keep the naming convention consistent with: "frr-image" as the flag?

yes makes sense, done

**Comment by fedepaol** [comment]
CI failure fixed by https://github.com/metallb/metallb/pull/1900

**Pull request merged**